tests/services/test_users.py

```diff
@@ -192,27 +266,76 @@ def test_get_available_roles_calls_get_with_expected_uri(self, mock_connection):
         uri = "/api/v1/role"
         mock_connection.get.assert_called_once_with(uri)
 
+    @patch.object(
+        py42.services.users.UserService,
+        "_get_user_uid_by_id",
+        return_value=TEST_USER_UID,
+    )
     def test_get_roles_calls_get_with_expected_uri(self, mock_connection):
+        user_id = 12345
         service = UserService(mock_connection)
-        service.get_roles(12345)
-        uri = "/api/v1/UserRole/12345"
-        mock_connection.get.assert_called_once_with(uri)
+        service.get_roles(user_id)
+        uri = f"{USER_URI_V3}/{TEST_USER_UID}/roles"
+        mock_connection.get.assert_called_with(uri)
 
-    def test_add_role_calls_post_with_expected_uri_and_data(self, mock_connection):
+    def test_get_role_ids_calls_get_roles_and_returns_role_ids(
+        self, mock_connection, mock_get_roles_response
+    ):
+        user_id = 12345
+
+        with patch.object(
+            py42.services.users.UserService,
+            "get_roles",
+            return_value=mock_get_roles_response,
+        ) as mock_get_roles:
+            service = UserService(mock_connection)
+            role_ids = service._get_role_ids(user_id)
+
+        mock_get_roles.assert_called_once_with(user_id)
+        assert role_ids == ["desktop-user", "proe-user", "customer-cloud-admin"]
+
+    @patch.object(
+        py42.services.users.UserService,
+        "_get_user_uid_by_id",
+        return_value=TEST_USER_UID,
+    )
+    def test_update_roles_calls_put_with_expected_uri(self, mock_connection):
         service = UserService(mock_connection)
-        service.add_role(12345, "Test Role Name")
-        uri = "/api/v1/UserRole"
-        assert mock_connection.post.call_args[0][0] == uri
-        assert mock_connection.post.call_args[1]["json"]["roleName"] == "Test Role Name"
-        assert mock_connection.post.call_args[1]["json"]["userId"] == 12345
+        role_ids = ["role-1", "role.name.2"]
+        service._update_roles(12345, role_ids)
+        uri = f"/api/v3/users/{TEST_USER_UID}/roles"
+        data = {"roleIds": role_ids}
+        mock_connection.put.assert_called_once_with(uri, json=data)
+
+    def test_update_role_ids_calls_get_available_roles_and_returns_role_ids_when_add(
```

Comment:
opted to unit test the helper methods rather than the full `add/remove_role` functions due to all the additional lookups necessary